internal/util-interface/src/main/java/xsbti/Problem.java

```diff
@@ -26,4 +28,24 @@ public interface Problem {
   default Optional<String> rendered() {
     return Optional.empty();
   }
+
+  /**
+   * The unique code attatched to the diagnostic being reported.
+   *
+   * <p>NOTE: To avoid breaking compatibility we provide a default to account for older Scala
+   * version that do not have codes.
+   */
+  default Optional<DiagnosticCode> diagnosticCode() {
```

Comment:
Is this enough information to get sbt to produce errors that look like what dotty natively produce without us having to pass in a `rendered` string? That would help solve https://github.com/lampepfl/dotty/issues/14691